notebooks/ch15/bias_cv.md

```diff
@@ -0,0 +1,471 @@
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>Validation Error</th>
+    </tr>
+    <tr>
+      <th>Degree</th>
+      <th></th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>1</th>
+      <td>0.332086</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>0.046725</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>0.047069</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>0.048161</td>
+    </tr>
+    <tr>
+      <th>5</th>
+      <td>0.048352</td>
+    </tr>
+    <tr>
+      <th>6</th>
+      <td>0.048653</td>
+    </tr>
+    <tr>
+      <th>7</th>
+      <td>0.048578</td>
+    </tr>
+    <tr>
+      <th>8</th>
+      <td>0.048877</td>
+    </tr>
+    <tr>
+      <th>9</th>
+      <td>0.051174</td>
+    </tr>
+    <tr>
+      <th>10</th>
+      <td>0.050455</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+We can see that as we use higher degree polynomial features, the validation error decreases and increases again.
+
+
+```python
+# HIDDEN
+plt.figure(figsize=(10, 5))
+
+plt.subplot(121)
+plt.plot(cv_df.index, cv_df['Validation Error'])
+plt.scatter(cv_df.index, cv_df['Validation Error'])
+plt.title('Validation Error vs. Polynomial Degree')
+plt.xlabel('Polynomial Degree')
+plt.ylabel('Validation Error');
+
+plt.subplot(122)
+plt.plot(cv_df.index, cv_df['Validation Error'])
+plt.scatter(cv_df.index, cv_df['Validation Error'])
+plt.ylim(0.046225, 0.05)
+plt.title('Zoomed In')
+plt.xlabel('Polynomial Degree')
+plt.ylabel('Validation Error')
+
+plt.tight_layout();
+```
+
+
+![png](bias_cv_files/bias_cv_22_0.png)
+
+
+Examining the validation errors reveals the most accurate model only used degree 2 polynomial features. Thus, we select the degree 2 polynomial model as our final model and fit it on the all of the training data at once. Then, we compute its error on the test set.
+
+
+```python
+best_trans = transformers[1]
+best_model = LinearRegression(fit_intercept=False).fit(X_train_polys[1], y_train)
+
+training_error = mse_cost(best_model.predict(X_train_polys[1]), y_train)
+validation_error = cross_validation_errors[1]
+test_error = mse_cost(best_model.predict(best_trans.transform(X_test)), y_test)
+
+print('Degree 2 polynomial')
+print(f'  Training error: {training_error:0.5f}')
+print(f'Validation error: {validation_error:0.5f}')
+print(f'      Test error: {test_error:0.5f}')
+```
+
+    Degree 2 polynomial
+      Training error: 0.04427
+    Validation error: 0.04673
+          Test error: 0.04745
+    
+
+For future reference, Scikit-learn has [`cross_val_predict`](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_predict.html) to automatically perform cross-validation, so you don't have to break the data into training and validation sets yourself. Also, note that the test error is higher than the validation error which is higher than the training error. 
```

Comment:
Nit: `scikit-learn` instead of "Scikit-learn"
First sentence: Maybe "has [a] cross_val_predict [method]"
First sentence: "[...] so [we] don't have to break the data into training and validation sets [ourselves.]"

The last sentence seems kind of tacked on. I would expand on this (maybe with an intuitive explanation of this phenomenon) and make this its own paragraph.